lib/features/mailbox/presentation/mailbox_controller.dart

```diff
@@ -1338,6 +1381,46 @@ class MailboxController extends BaseMailboxController with MailboxActionHandlerM
     }
   }
 
+  void _handleSubaddressingAction(MailboxId mailboxId, Map<String, List<String>?>? currentRights, MailboxActions subaddressingAction) {
+    final accountId = mailboxDashBoardController.accountId.value;
+    final session = mailboxDashBoardController.sessionCurrent;
+
+    if (session != null && accountId != null) {
+      final allowSubaddressingRequest = MailboxRightRequest(
+          mailboxId,
+          currentRights,
+          subaddressingAction == MailboxActions.allowSubaddressing ? MailboxSubaddressingAction.allow : MailboxSubaddressingAction.disallow
+      );
+
+      consumeState(_subaddressingInteractor.execute(session, accountId, allowSubaddressingRequest));
+    } else {
+      _handleSubaddressingFailure(SubaddressingFailure.withException(NullSessionOrAccountIdException()));
+    }
+
+    popBack();
+  }
+
+  void _handleSubaddressingFailure(SubaddressingFailure failure) {
+    if (currentOverlayContext != null && currentContext != null) {
+      final messageError = AppLocalizations.of(currentContext!).toastMessageSubaddressingFailure;
+      appToast.showToastErrorMessage(currentOverlayContext!, messageError);
+    }
+  }
+
+  void _handleSubaddressingSuccess(SubaddressingSuccess success) {
+    appToast.showToastSuccessMessage(
+      currentOverlayContext!,
+      success.subaddressingAction == MailboxSubaddressingAction.allow
+          ? AppLocalizations.of(currentContext!).toastMessageAllowSubaddressingSuccess
+          : AppLocalizations.of(currentContext!).toastMessageDisallowSubaddressingSuccess,
+    );
+
+    _refreshMailboxChanges(
+        currentMailboxState: success.currentMailboxState,
+        properties: MailboxConstants.propertiesDefault
+    );
```

Comment:
IMO, it is an unnecessary call. Duplicated refresh request.

<img width="497" alt="Screenshot 2024-11-28 at 11 47 47" src="https://github.com/user-attachments/assets/718b5967-5c37-4693-9066-da0ef0d39803">